docs/devices-api/more-info/webhooks.mdx

```diff
@@ -17,6 +17,21 @@ how the Webhook flow works, see [our Webhooks guide](/home/api-info/webhooks).
   page](/devices-api/more-info/our-integrations).
 </Info>
 
+## User provider subscription
+
+The format follows:
+
+```json
+{
+  "meta": {
+    "messageId": "<message-id>",
+    "when": "<date-time-in-utc>"
+  },
+  "userId": "951faef1-5cfd-464a-81f7-31f76edf309e",
+  "connectedProviders": ["fitbit", "withings", "cronometer"]
+}
```

Comment:
I suggest we keep the same structure we have for the providers' data, something like:

```json
{
  "meta": {
    "messageId": "<message-id>",
    "when": "<date-time-in-utc>"
  },
  "users": [{
     "userId": "951faef1-5cfd-464a-81f7-31f76edf309e",
     "connectedProviders": ["fitbit", "withings", "cronometer"],
  }]
}
```

Reply:
Even though this will always happen for one user at a time? Because with the current implementation, we're not grouping this information coming from different users

Reply:
I'm looking at this from our customer's perspective. If we were to parse this, what would be the best format?
1. We don't have a "type" prop on the message, so to parse it one would need to check different formats, with `userId` on the array and out of the array;
2. Even though we're getting only one element in the array, we already have code to parse WH payloads with an array of users, so keeping the same logic makes it easier;

From Metriport's perspective, now we're sending one message per user per subscription, but with the array of users we could, in the future, group all subscriptions that came happened in the last X minutes if we so choose (to deal with system overload issues, for example). This is not a reason on itself (since this is "building for the future"), but it's a benefit that we get from this approach nonetheless.

Reply:
Makes sense! Foresight + customer POV. Ty